### PR TITLE
Document custom options extraCost update

### PR DESCRIPTION
**Pull request by dpobel**
See https://gitlab.com/front-commerce/front-commerce/-/merge_requests/622

### CODE REVIEW

File: source/docs/appendices/migration-guides.md [hunk 1]
```diff
@@ -9,13 +9,22 @@ Our goal is to make migrations as smooth as possible. This is why we try to make
...
+- `theme/modules/ProductView/withSelectedCustomOptions.js`
+
+So if you have overridden those files, you have to apply [the same update than the one done in Front-Commerce](https://gitlab.com/front-commerce/front-commerce/-/commit/d3dbac21907cb2eae21cef7876bcbdd7b642e66f?merge_request_iid=622).
 
 ### New features in `2.9.0`
 
 These new features may be relevant for your existing application:
 
-- no new feature yet!
+- For custom options, the Graph now exposes an `extraCost` field where you can find the rate to apply or the prices including and excluding taxes of each option and value.
```
Comment (dpobel) [inline source/docs/appendices/migration-guides.md:27]: self note: check that the commit hash is the right one before merging this

File: source/docs/appendices/migration-guides.md [hunk 2]
```diff
@@ -9,13 +9,22 @@ Our goal is to make migrations as smooth as possible. This is why we try to make
...
+### Customs options `price` and `price_type` field deprecations
+
+As of the version 2.9.0, the `price` and `price_type` fields of the types `ProductCustomOption` and `ProductCustomOptionValue` have been deprecated in order to provide both the price including and excluding taxes for each options. Custom options prices or rate are now carried by the `extraCost` field available in those types. To stop using those deprecated fields, the following files have been modified:
+
+- `theme/modules/Cart/CartItem/CartItemOptionsUpdater/CartItemOptionsUpdaterFragment.gql`
+- `theme/components/organisms/Configurator/ProductConfiguratorFragment.gql`
+- `theme/components/organisms/Configurator/CustomOption/CustomOptionLabelWithPrice.js`
+- `theme/modules/ProductView/withSelectedCustomOptions.js`
+
+So if you have overridden those files, you have to apply [the same update than the one done in Front-Commerce](https://gitlab.com/front-commerce/front-commerce/-/commit/d3dbac21907cb2eae21cef7876bcbdd7b642e66f?merge_request_iid=622).
```
Comment (dpobel) [inline source/docs/appendices/migration-guides.md:21]: self note: update the commit hash before after the review and before merging this PR

**Review by real34 — APPROVED** [review]

File: source/docs/appendices/migration-guides.md [hunk 3]
```diff
@@ -9,13 +9,22 @@ Our goal is to make migrations as smooth as possible. This is why we try to make
...
+### Customs options `price` and `price_type` field deprecations
+
+As of the version 2.9.0, the `price` and `price_type` fields of the types `ProductCustomOption` and `ProductCustomOptionValue` have been deprecated in order to provide both the price including and excluding taxes for each options. Custom options prices or rate are now carried by the `extraCost` field available in those types. To stop using those deprecated fields, the following files have been modified:
+
+- `theme/modules/Cart/CartItem/CartItemOptionsUpdater/CartItemOptionsUpdaterFragment.gql`
+- `theme/components/organisms/Configurator/ProductConfiguratorFragment.gql`
+- `theme/components/organisms/Configurator/CustomOption/CustomOptionLabelWithPrice.js`
+- `theme/modules/ProductView/withSelectedCustomOptions.js`
+
+So if you have overridden those files, you have to apply [the same update than the one done in Front-Commerce](https://gitlab.com/front-commerce/front-commerce/-/commit/79c69a9d294dd58979dbd5d94153edf9e29b3d28?merge_request_iid=622).
```
Comment (real34) [inline source/docs/appendices/migration-guides.md:21]:
```suggestion
So if you have overridden those files, you have to apply [the same update than the one done in Front-Commerce](https://gitlab.com/front-commerce/front-commerce/-/commit/79c69a9d294dd58979dbd5d94153edf9e29b3d28?merge_request_iid=622).

We highly recommend you to test your product and cart page for products with custom options of both "fixed price" and "percentage" types if your projects makes use of this Magento feature.
```

I feel it's good to insist on this one after reading the code changes related to this feature. We're never too safe.

**Review by real34 — APPROVED** [review]

File: source/docs/appendices/migration-guides.md [hunk 4]
```diff
@@ -9,13 +9,22 @@ Our goal is to make migrations as smooth as possible. This is why we try to make
 
 ## `2.8.0` -> `2.9.0`
 
-No action needed for migrating to this version.
+### Customs options `price` and `price_type` field deprecations
```
[the remark this answers is not in the document]
Comment (dpobel) [reply source/docs/appendices/migration-guides.md:12]: actually for Magento2, the 2.4.0 won't be enough as the patch requires https://gitlab.com/front-commerce/magento2-module-front-commerce/-/merge_requests/89 which is not merged yet. Will that result in a 2.5 version ?

File: source/docs/appendices/migration-guides.md (hunk 3 above)
Comment (dpobel) [reply source/docs/appendices/migration-guides.md]: done

File: source/docs/appendices/migration-guides.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dpobel) [reply source/docs/appendices/migration-guides.md:20]: (add a note about the Magento module requirement, assuming the next version of the Magento2 module will be the 2.5.0)

[the remark this answers is not in the document]
Comment (real34) [reply source/docs/appendices/migration-guides.md:20]: Yes, perfect. It will be a 2.5.0 version.

**Review by real34 — APPROVED** [review]

**Pull request merged**